### PR TITLE
[CDS-51341] Filter support for templates

### PR DESCRIPTION
**Pull request by rashminandasahoo**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by utkarshchoubey-harness — DISMISSED** [review]

File: docs/platform/13_Templates/template.md [hunk 1]
```diff
@@ -226,3 +226,32 @@ template:
 ```
 The YAML editor is a full-fledged YAML IDE with autocomplete and other features. See [Harness YAML Quickstart](../8_Pipelines/harness-yaml-quickstart.md).
 
+### Template filter
+
+You can create your own template filter and use it later to view those templates in Harness. You can create template filter at any scope.
```
Comment (StrawberryPad) [inline docs/platform/13_Templates/template.md:231]:
```suggestion
You can create your own template filter and use it later to view those templates in Harness. You can create a template filter at any scope.
```

File: docs/platform/13_Templates/template.md [hunk 2]
```diff
@@ -226,3 +226,32 @@ template:
...
+This topic explains steps to create a template filter at the account scope.
+To create a template filter in Harness: 
+1. Select **ACCOUNT SETTINGS** and then select **Account Resources**.
```
Comment (StrawberryPad) [inline docs/platform/13_Templates/template.md:234]:
```suggestion
1. Select **ACCOUNT SETTINGS**, and then select **Account Resources**.
```

File: docs/platform/13_Templates/static/template-new-filter-settings.png
Comment (StrawberryPad) [inline docs/platform/13_Templates/static/template-new-filter-settings.png:1]: This screen shot should be cropped to a third of its size... you only need the top part, no need for the empty space to the bottom.

File: docs/platform/13_Templates/template.md [hunk 3]
```diff
@@ -226,3 +226,32 @@ template:
...
+2. Select **Templates**.
+3. In the top right corner, select the filter icon.
+   
+   ![](./static/template-filter-icon.png)
+
+   **New Filter** settings appear.
+4. In **Name** enter the name of an existing template.
```
Comment (StrawberryPad) [inline docs/platform/13_Templates/template.md:241]:
```suggestion
4. In **Name** enter the name of an existing template, and then select **New Filter**.
```

File: docs/platform/13_Templates/template.md [hunk 4]
```diff
@@ -226,3 +226,32 @@ template:
...
+   
+   ![](./static/template-new-filter-settings.png)
+   
+5. Select **New Filter** on the right.
```
Comment (StrawberryPad) [inline docs/platform/13_Templates/template.md:245]:
```suggestion
```

File: docs/platform/13_Templates/template.md [hunk 5]
```diff
@@ -226,3 +226,32 @@ template:
...
+6. In **Filter Name**, enter a name for the filter.
+7. You can set the visibility of the filter by selecting one of the following:
```
Comment (StrawberryPad) [inline docs/platform/13_Templates/template.md:247]:
```suggestion
7. Set the visibility of the filter by selecting one of the following: 
```

File: docs/platform/13_Templates/template.md [hunk 6]
```diff
@@ -226,3 +226,32 @@ template:
...
+4. In **Name** enter the name of an existing template.
+   
+   ![](./static/template-new-filter-settings.png)
+   
+5. Select **New Filter** on the right.
+6. In **Filter Name**, enter a name for the filter.
+7. You can set the visibility of the filter by selecting one of the following: 
+   - **Only me**
+   - **Everyone**
+   If you do not select anything, the visibility is set to **Only me** by default.
```
Comment (StrawberryPad) [inline docs/platform/13_Templates/template.md:250]:
```suggestion

   If you do not select anything, the visibility is set to **Only me** by default.
```

Comment (StrawberryPad) [reply docs/platform/13_Templates/template.md:250]: space so it doesn't show up on the same line as **Everyone**

File: docs/platform/13_Templates/template.md [hunk 7]
```diff
@@ -226,3 +226,32 @@ template:
...
+
+   ![](./static/template-filter-config.png)
+
+8. Select **Save**.
+9. Select **Apply**.
```
Comment (StrawberryPad) [inline docs/platform/13_Templates/template.md:255]:
```suggestion
8. Select **Save**, and then **Apply**.

```

File: docs/platform/13_Templates/template.md
Comment (StrawberryPad) [inline docs/platform/13_Templates/template.md:1]: General comment: This topic has only h3's (###) - at some point they should all be changed to h2s (##) as normal.

**Review by StrawberryPad — CHANGES_REQUESTED** [review]
These are mostly "should do" to improve usability. There's only one typo (missing an "a" in a sentence).

**Review by StrawberryPad — APPROVED** [review]

**Pull request merged**